tests/unit/TechRecordsService.unitTest.ts

```diff
@@ -410,6 +416,124 @@ describe("insertTechRecord", () => {
+        it("should return error 500 TrailerId generation failed", async () => {
+          const MockDAO = jest.fn().mockImplementation(() => {
+            return {
+              getTrailerId: () => {
+                return Promise.resolve({
+                  trailerLetter: "C",
+                  sequenceNumber: 530001,
+                  testNumberKey: 2
+                });
+              }
+            };
+          });
+          const mockDAO = new MockDAO();
+          const techRecordsService = new TechRecordsService(mockDAO);
+
+          // @ts-ignore
+          const techRecord: ITechRecordWrapper = cloneDeep(records[78]);
+          techRecord.vin = Date.now().toString();
+          delete techRecord.trailerId;
+          delete techRecord.techRecord[0].statusCode;
+
+          try {
+            expect(await techRecordsService.setTrailerId(techRecord)).toThrowError();
+          } catch (errorResponse) {
+            expect(errorResponse.statusCode).toEqual(500);
+            expect(errorResponse.body).toEqual(ERRORS.TrailerIdGenerationFailed);
+          }
+        });
+      });
+
+      context("and the trailer id object contains the error attribute", () => {
+        it("should return error 500 TrailerId generation failed", async () => {
+          const MockDAO = jest.fn().mockImplementation(() => {
+            return {
+              getTrailerId: () => {
+                return Promise.resolve({
+                  error: "Some error from test-number microservice"
+                });
+              }
+            };
+          });
+          const mockDAO = new MockDAO();
+          const techRecordsService = new TechRecordsService(mockDAO);
+
+          // @ts-ignore
+          const techRecord: ITechRecordWrapper = cloneDeep(records[78]);
+          techRecord.vin = Date.now().toString();
+          delete techRecord.trailerId;
+          delete techRecord.techRecord[0].statusCode;
+
+          try {
+            expect(await techRecordsService.setTrailerId(techRecord)).toThrowError();
+          } catch (errorResponse) {
+            expect(errorResponse.statusCode).toEqual(500);
+            expect(errorResponse.body).toEqual("Some error from test-number microservice");
+          }
+        });
+      });
+
+      context("and the test-number microservice returned an error", () => {
+        it("should return error 500", async () => {
+          const MockDAO = jest.fn().mockImplementation(() => {
+            return {
+              getTrailerId: () => {
+                return Promise.reject("Error from test-number microservice");
+              }
+            };
+          });
+          const mockDAO = new MockDAO();
+          const techRecordsService = new TechRecordsService(mockDAO);
+
+          // @ts-ignore
+          const techRecord: ITechRecordWrapper = cloneDeep(records[78]);
+          techRecord.vin = Date.now().toString();
```

Comment:
What is happening here?

Reply:
test case when the call to test-number microservice to generate a new trailerId failed.

Reply:
If you're referring to "techRecord.vin = Date.now().toString();" I think I might have copied the test-case from an integration test and forgot to delete this line

Reply:
deleted all the unnecessary lines

Reply:
Yes, I was referring to that line. Thanks for removing